### ISSUE TITLE
Empty list connections return "null" instead of empty array.

### ISSUE THREAD

**Issue by ChrisWiegman** [wp-graphql/wp-graphql#2144]
When creating a connection to list a list of posts GraphQL returns "null" instead of an empty array if no posts are found. This is different from standard query returns where an empty array is displayed.

The easiest way to recreate is probably by using ACM to create a related field and then query for nodes as appropriate. See the screenshots below for a demonstration of the issue

![A normal query when no content is found](https://user-images.githubusercontent.com/394675/140522427-bf6e10c4-1cb8-4fbf-a139-3ad6c6dda8db.png)

![A connection query where no content is found. Note the "null" instead of an empty array](https://user-images.githubusercontent.com/394675/140522440-207e9ecf-ae63-42c2-98ff-1d36b90e22bc.png)

**Comment by jasonbahl** [comment]
@ChrisWiegman without steps to create models the same way you have them created I had to do a bit of guess work. 

I tried a few different ways of establishing connections between models, and ultimately what I discovered is that the difference in behavior is actually right here in front of us! 

It looks like the Cage to Rabbit Connection in the 2nd screenshot is a one-to-one connection. 

The Cage can have One rabbit, which is why it's queried with `node` instead of `nodes`, and why it either returns a rabbit or null. 

The first screenshot is a plural connection, which is why it's queried with `nodes` and returns either a populated list or an empty list.
